addons/sale_project/models/sale_order.py

```diff
@@ -155,6 +157,31 @@ def _auto_init(self):
             """)
         return super()._auto_init()
 
+    @api.depends('product_id')
+    def _compute_qty_delivered_method(self):
+        super(SaleOrderLine, self)._compute_qty_delivered_method()
+        for line in self:
+            if not line.is_expense and line.product_id.type == 'service' and line.product_id.service_type == 'milestones':
+                line.qty_delivered_method = 'milestones'
+
+    @api.depends('reached_milestones_ids.quantity_percentage')
```

Comment:
I think we could also add the field that we need in this method even if those fields are in the depends in the parent method(s) because if one day the depends change in the parent one, we could miss a trigger for this method. What do you think? :) 
```suggestion
    @api.depends('qty_delivered_method', 'product_uom_qty', 'reached_milestones_ids.quantity_percentage')
```

Reply:
You are right, I forgot to add them. Fixed :slightly_smiling_face: